packages/core/src/controllers/relayer.ts

```diff
@@ -62,20 +64,22 @@ export class Relayer extends IRelayer {
       typeof opts.logger !== "undefined" && typeof opts.logger !== "string"
         ? generateChildLogger(opts.logger, this.name)
         : pino(getDefaultLoggerOptions({ level: opts.logger || RELAYER_DEFAULT_LOGGER }));
-    const rpcUrl =
-      opts.rpcUrl ||
-      formatRelayRpcUrl(this.protocol, this.version, RELAYER_DEFAULT_RELAY_URL, opts.projectId);
-    this.provider =
-      typeof opts.relayProvider !== "string" && typeof opts.relayProvider !== "undefined"
-        ? opts.relayProvider
-        : new JsonRpcProvider(new WsConnection(rpcUrl));
     this.messages = new MessageTracker(this.logger, opts.core);
     this.subscriber = new Subscriber(this, this.logger);
     this.publisher = new Publisher(this, this.logger);
+
+    // re-assigned during init()
+    this.providerOpts = { ...opts };
+    this.provider = {} as IJsonRpcProvider;
```

Comment:
Mainly to avoid checking for provider being undefined everywhere where it is used (a lot in relay). It becomes "defined" after initialisation and we already have guards for that